docs/get-started/extensions.qmd

```diff
@@ -0,0 +1,170 @@
+---
+title: Extending Great Tables
+format: html
+---
+
+<!-- Intro -->
+The **Great Tables** package boasts a diverse range of methods for creating beautiful tables. With no shortage of tools, styles, formats and themes, you can produce a very wide range of tables. If you haven't yet, head over to the [examples gallery](../examples/index.qmd) for a flavor of the variety that is possible.
+
+But what happens when there's something you want to add to your table, and it's not directly supported by **Great Tables**? Let's say you have a very specific use case for your great table, so it's not necessarily broad enough to [contribute to the package](contributing.qmd). Or maybe it *is* generalizable, but it's beyond the scope of what you want to do. Not every feature fits neatly into a pull request.
+
+[**gt-extras**](https://posit-dev.github.io/gt-extras/) is a great case study in this, filling some table goals that aren't supported by **Great Tables** directly.
+
+## Writing an Extension Function
+
+When extending **Great Tables**, you have three main approaches: *wrappers*, *extensions*, and what we might call *composing functions*.
+
+- **Wrappers** simplify existing functionality by bundling common parameter combinations into convenient functions. They don't add new capabilities, but make complex styling or formatting tasks more accessible.
+- **Composing functions** combine multiple methods into a single function, allowing you to build more complex table logic or reusable patterns.
+- **Extensions**, on the other hand, add new functionality by combining multiple methods in creative ways, processing data outside the package, or generating custom content (like HTML or images) that gets integrated into your table.
+
+All approaches are valuable contributions to the **Great Tables** ecosystem, and the choice between them depends on whether you're making it easier to access existing tooling or creating entirely new capabilities.
+
+### Wrappers
+
+Let's wrap `~~GT.tab_style()` to perform a specific action supported by the general function. `~~GT.tab_style()` takes `locations=` and `style=` parameters, so let's assign those. Something this lightweight might be useful for someone who knows what effect they want, but doesn't want to investigate the entire space of options for styling.
+
+```{python}
```

Comment:
It took me a second to find where columns was used. I might cut the example down so it's quick to scan... (keeping two locations makes sense to me b/c it's sort of wrapping the user input...):

```{python}
def highlight_cols(gt: GT, columns: SelectExpr) -> GT:
    locations = [
        loc.body(columns=columns),
        loc.column_labels(columns=columns),
    ]

    # Make the wrapped tab_style call
    return gt.tab_style(
        style=style.fill(color="lightblue"),
        locations=locations,
    )
```